From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Add missing standard includes to vendored grpc for clang

grpc 1.74.0 omits standard includes that clang requires: glob.cc needs
<algorithm> (std::any_of) and event_engine/memory_request.h needs
<string> (std::to_string). Prepend them via patch_cmds.
---
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index efc4bdeb..b05dca93 100644
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -495,6 +495,11 @@ def _tf_repositories():
         patch_file = [
             "@xla//third_party/grpc:grpc.patch",
         ],
+        # grpc 1.74.0 headers/sources omit standard includes that clang requires.
+        patch_cmds = [
+            "sed -i '1i #include <algorithm>' src/core/util/glob.cc",
+            "sed -i '1i #include <string>' include/grpc/event_engine/memory_request.h",
+        ],
         urls = tf_mirror_urls("https://github.com/grpc/grpc/archive/refs/tags/v1.74.0.tar.gz"),
     )
 
